spyrit/learning/model_Had_DCAN.py

```diff
@@ -834,8 +835,8 @@ def forward(self, x):
         return x
     
     def forward_denoise(self, x, var, b, c, h, w):
```

Comment:
Add `forward_denoise_expe(self, x, var, N0_est b, c, h, w)` where `N0_est` is `(b,c)`